core/src/main/scala/chisel3/Module.scala

```diff
@@ -116,17 +116,22 @@ object Module extends SourceInfoDoc {
     *  This assigns the explicit directions required by both semantics on all Bundles.
     * This recursively walks the tree, and assigns directions if no explicit
     *   direction given by upper-levels (override Input / Output)
+    * As this is for compatibility, only Data is considered, and Properties are skipped.
     */
-  private[chisel3] def assignCompatDir(data: Data): Unit =
-    // Collect all leaf elements of the data which have an unspecified or flipped
-    // direction, and assign explicit directions to them
-    DataMirror
-      .collectMembers(data) {
-        case x: Element
-            if x.specifiedDirection == SpecifiedDirection.Unspecified || x.specifiedDirection == SpecifiedDirection.Flip =>
-          x
-      }
-      .foreach { x => x._assignCompatibilityExplicitDirection }
+  private[chisel3] def assignCompatDir(data: IsBindable): Unit = {
+    data match {
+      case (d: Data) =>
```

Comment:
I'm not sure why this is needed, but since it's about compatibility, I'm assuming we don't need to do this for things which aren't Data (e.g., new things like Property).

Reply:
Since this is only called in two places that are private APIs and are statically known to be `Data`, I think we can just leave this as being defined on `Data` and update the call sites.

Reply:
Actually, looking at this again, assignCompatDir is used by ClonePorts, where we know we're working with Data, but it's also used in bindIoInPlace, which is used by `IO.apply` for ports. And I did want `IO.apply` to work with `IsBindable`. So I guess the idea is assignCompatDir takes a Data, which isn't a problem with ClonePorts, and in bindIoInPlace, it will only call assignCompatDir if the bindable is a Data?